### PR TITLE
Adapted to Swift 3.0

### PR DESCRIPTION
**Pull request by jonohayon**
I've adapted the library to swift 3, just some small changes to what Xcode told me to change and what I knew I needed to change (such as the `dispatch_global` to `DispatchQueue.global` and some other stuff).

Also added syntax highlighting to the README.md.

### CODE REVIEW

**Comment by daniel-pedersen** [comment]
Thank you rezozo!

However, there are some things that need to be changed before I can merge. You can check my comments.

File: SKQueue.swift [hunk 1]
```diff
@@ -165,27 +165,27 @@ class SKQueue {
...
-                    self.delegate?.receivedNotification(self, notification, forPath: pathEntry.path)
+                DispatchQueue.main.async {
+                  self.delegate?.receivedNotification(queue: self, notification, forPath: pathEntry.path)
                 }
             }
         }
         
         if close(fd) == -1 {
-            NSLog("SKQueue watcherThread: Couldn't close main kqueue (%d)", errno)
+            print("SKQueue watcherThread: Couldn't close main kqueue (\(errno))")
```
Comment (daniel-pedersen) [inline SKQueue.swift:177]: `NSLog` is not interchangeable with `print` (NSLog writes to the system console). This should be left as-is.

File: SKQueue.swift [hunk 2]
```diff
@@ -6,26 +6,26 @@
...
+//  including commercial applications, and to alter it and redistribute it
+//  freely, subject to the following restrictions:
+//     1. The origin of this software must not be misrepresented; you must not
+//     claim that you wrote the original software. If you use this software
+//     in a product, an acknowledgment in the product documentation would be
+//     appreciated but is not required.
+//     2. Altered source versions must be plainly marked as such, and must not be
+//     misrepresented as being the original software.
+//     3. This notice may not be removed or altered from any source
+//     distribution.
```
Comment (daniel-pedersen) [inline SKQueue.swift:22]: This should not be changed in this PR. I'll fix it during a future style cleanup.

File: README.md [hunk 3]
```diff
@@ -4,20 +4,26 @@ Lightweight port of [VDKQueue](https://github.com/bdkjones/VDKQueue).
...
 
 ## Example
-    class SomeClass: SKQueueDelegate {
-        func receivedNotification(queue: SKQueue, _ notificationName: SKQueueNotificationString, forPath path: String) {
-            print("\(notificationName.rawValue) @ \(path)")
-        }
+```swift
+class SomeClass: SKQueueDelegate {
+    func receivedNotification(queue: SKQueue, _ notificationName: SKQueueNotificationString, forPath path: String) {
+        print("\(notificationName.rawValue) @ \(path)")
```
Comment (daniel-pedersen) [inline README.md:10]: Awesome with the syntax highlighting. I would prefer you do a separate PR with these changes though, just to keep the commit history as clean as possible.

**Review by daniel-pedersen — DISMISSED** [review]

**Comment by schlemiel29** [comment]
I used the old version and it worked fine. After updating SKQueue with these changes (Swift 3.0), nothing happens anymore. Is there a small test app, where I can check what happens, when a file is updated, removed, copied etc...?

**Comment by daniel-pedersen** [comment]
Threading is a bit different with the new DispatchQueue API, which is why this version breaks. I have now added working Swift 3.0 support on master.
